R-package/R/utils.R

```diff
@@ -535,44 +529,88 @@ NULL
 NULL
 
 # Lookup table for the deprecated parameters bookkeeping
-depr_par_lut <- matrix(c(
-  'print.every.n', 'print_every_n',
-  'early.stop.round', 'early_stopping_rounds',
-  'training.data', 'data',
-  'with.stats', 'with_stats',
-  'numberOfClusters', 'n_clusters',
-  'features.keep', 'features_keep',
-  'plot.height', 'plot_height',
-  'plot.width', 'plot_width',
-  'n_first_tree', 'trees',
-  'dummy', 'DUMMY',
-  'watchlist', 'evals'
-), ncol = 2, byrow = TRUE)
-colnames(depr_par_lut) <- c('old', 'new')
+deprecated_train_params <- list(
+  'print.every.n' = 'print_every_n',
+  'early.stop.round' = 'early_stopping_rounds',
+  'training.data' = 'data',
+  'dtrain' = 'data',
+  'watchlist' = 'evals',
+  'feval' = 'custom_metric'
+)
+deprecated_dttree_params <- list(
+  'n_first_tree' = 'trees'
+)
+deprecated_plot_params <- list(
+  'plot.height' = 'plot_height',
+  'plot.width' = 'plot_width'
+)
+deprecated_multitrees_params <- c(
+  deprecated_plot_params,
+  list('features.keep' = 'features_keep')
+)
+deprecated_dump_params <- list(
+  'with.stats' = 'with_stats'
+)
+deprecated_plottree_params <- c(
+  deprecated_plot_params,
+  deprecated_dump_params
+)
 
 # Checks the dot-parameters for deprecated names
 # (including partial matching), gives a deprecation warning,
 # and sets new parameters to the old parameters' values within its parent frame.
 # WARNING: has side-effects
-check.deprecation <- function(..., env = parent.frame()) {
-  pars <- list(...)
-  # exact and partial matches
-  all_match <- pmatch(names(pars), depr_par_lut[, 1])
-  # indices of matched pars' names
-  idx_pars <- which(!is.na(all_match))
-  if (length(idx_pars) == 0) return()
-  # indices of matched LUT rows
-  idx_lut <- all_match[idx_pars]
-  # which of idx_lut were the exact matches?
-  ex_match <- depr_par_lut[idx_lut, 1] %in% names(pars)
-  for (i in seq_along(idx_pars)) {
-    pars_par <- names(pars)[idx_pars[i]]
-    old_par <- depr_par_lut[idx_lut[i], 1]
-    new_par <- depr_par_lut[idx_lut[i], 2]
-    if (!ex_match[i]) {
-      warning("'", pars_par, "' was partially matched to '", old_par, "'")
+check.deprecation <- function(
```

Comment:
Is there any test for this function?

Reply:
The function is not tested directly as it uses the parent frame, but the tests currently there check that the function does what it is supposed to when called from inside `xgb.train`.

For example, if `evals` is not passed (either through `evals` or through `watchlist` which this function is automatically swapping), there would be no evaluation log left in the booster object.